### PR TITLE
demo(datepicker): create demos for datepicker a11y use cases

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/demo-app/a11y/datepicker/datepicker-a11y.html [hunk 1]
```diff
@@ -0,0 +1,112 @@
+<section>
+  <h2>Choose a date (e.g. choose your date of birth)</h2>
+  <md-input-container>
+    <input mdInput
```
Comment (jelbourn) [inline src/demo-app/a11y/datepicker/datepicker-a11y.html]: This input needs either a `<label>` or an `aria-label`
(same for other inputs that don't have one)

File: src/demo-app/a11y/datepicker/datepicker-a11y.html [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+           [mdDatepicker]="birthdayPicker"
+           [(ngModel)]="birthday"
+           #birthdayModel="ngModel"
+           [max]="maxBirthday"
+           required
+           placeholder="Date of birth">
+    <button mdSuffix [mdDatepickerToggle]="birthdayPicker"></button>
```
Comment (jelbourn) [inline src/demo-app/a11y/datepicker/datepicker-a11y.html]: Add a comment here like
```html
<!-- Button aria-label is added automatically by mdDatepickerToggle -->
```

File: src/demo-app/a11y/datepicker/datepicker-a11y.html (hunk 1 above)
Comment (mmalerba) [inline src/demo-app/a11y/datepicker/datepicker-a11y.html]: label is automatically created from the placeholder

**Comment by mmalerba** [comment]
PTAL

**Pull request merged**